slackminion/slack/conversation.py

```diff
@@ -7,20 +7,24 @@ class SlackConversation(object):
     _topic = None
 
     def __init__(self, conversation, api_client):
-        """Base class for rooms (channels, groups) and IMs"""
+        """
+        Base class for rooms (channels, groups) and IMs
+        """
         self.api_client = api_client
         self.conversation = conversation  # the dict slack sent us
         if conversation:
             self._topic = conversation.get('topic', {}).get('value')
         self.logger = logging.getLogger(type(self).__name__)
         self.logger.setLevel(logging.DEBUG)
 
+    # make arbitrary keys in the dict returned by slack accessible as properties
     def __getattr__(self, item):
         return self.conversation.get(item)
 
     @property
     def all_names(self):
-        return [self.name, self.conversation.get('normalized_name')] + self.conversation.get('previous_names', [])
+        return [self.conversation['name'], self.conversation['name_normalized']] \
+               + self.conversation.get('previous_names', [])
```

Comment:
Just to confirm I understand this correctly, the bug here was name_normalized is what is returned by the slack API and not normalized_name?

Reply:
No, this was not the bug, but we were using the wrong key for name_normalized.

Reply:
Would you mind helping educate me by pointing out the bug to me, as I can't seem to find it on my own. Thank you.

Also thanks for the tests!

Reply:
I haven't found the actual bug yet, I'm still working on that.